src/interface/swarm_device_context.hpp

```diff
@@ -60,10 +60,15 @@ class SwarmDeviceContext {
     int j = static_cast<int>(std::floor((y - y_min_) / ((y_max_ - y_min_) / 2.))) + 1;
     int k = static_cast<int>(std::floor((z - z_min_) / ((z_max_ - z_min_) / 2.))) + 1;
 
-    // Something went wrong
+    // Particle is on neither this block nor a neighboring block
     if (i < 0 || i > 3 || ((j < 0 || j > 3) && ndim_ > 1) ||
         ((k < 0 || k > 3) && ndim_ > 2)) {
-      PARTHENON_FAIL("Particle neighbor indices out of bounds");
+      printf("[%i] k = %i j = %i i = %i\n", n, k, j, i);
+      printf("x = %e [%e %e]\n", x, x_min_, x_max_);
+      printf("y = %e [%e %e]\n", y, y_min_, y_max_);
+      printf("z = %e [%e %e]\n", z, z_min_, z_max_);
+      PARTHENON_FAIL("Particle neighbor indices out of bounds; particle has somehow "
+                     "moved beyond the halo of adjacent blocks which is not permitted.");
```

Comment:
Are these print statements for testing or should they stay?

Reply:
This is a unique routine in that when anything goes wrong in particle transport in downstream applications, it is almost always caught by this check. So I wanted to make the output unusually verbose to speed up debugging.